pkg/rtc/rtc_linux_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package rtc
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+func TestSet(t *testing.T) {
+	for _, tt := range []struct {
+		name string
+	}{
+		{
+			name: "success",
+		},
+		{
+			name: "error",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			f, err := ioutil.TempFile("", "rtc-*")
+			if err != nil {
+				t.Error(err)
+			}
+			devs = []string{f.Name()}
+			f.Close()
+			rtc, err := OpenRTC()
+			if err != nil {
+				t.Error(err)
+			}
+			if tt.name == "success" {
+				unixIoctlSetRTCTime = func(fd int, value *unix.RTCTime) error {
+					return nil
+				}
+				err = rtc.Set(time.Now())
+				if err != nil {
+					t.Error(err)
+				}
+			}
+			if tt.name == "error" {
+				unixIoctlSetRTCTime = func(fd int, value *unix.RTCTime) error {
+					return errors.New("error")
+				}
+				err = rtc.Set(time.Now())
+				if err == nil {
+					t.Error(err)
+				}
+			}
+			rtc.Close()
+			err = os.Remove(devs[0])
```

Comment:
A common pattern is
if err := os.Remove(...); err != nil {
}

Reply:
I saw that comment a few times now and I wondered what the reason behind that pattern is. Don't get me wrong, I know it works but why deviate from a uniform pattern only in a few random occasions? Maybe you know something I don't so would be good to know the reason for that pattern.

Reply:
The reason for the pattern is to hold the err value for that one particular test. The err variable is captured in that one if err := block.  It avoids creating a variable outside the scope of that one particular function call, which can be useful for avoiding accidents (which have happened in practice) either caused by assigning to an err value that already existed; or by creating an err value that causes trouble later. It's a very common pattern, and it's a bit jarring not to see it used, as on lines 43 and 52; if it is ok it would be nice to use it. There is a writeup from the early go days about why to use it but ... I can't find it!

Reply:
Ah that makes sense! I'll change it up then, thanks for the explanation.